cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/cloudstorage/CloudStorageStreamSession.java

```
@@ -233,40 +244,59 @@ void sendBundle(Bundle bundle, boolean hasRefreshedCredentials)
                 throw new RuntimeException(e);
             }
         }
-        CreateSliceRequestPayload slicePayload = toCreateSliceRequestPayload(bundleStorageObject);
-        // Create slices on all replicas; remove the _failed_ replica if operation fails
-        replicas.removeIf(replica -> !tryCreateRestoreSlicePerReplica(replica, slicePayload));
-        if (!replicas.isEmpty())
+        createRestoreSliceOnSidecar(bundleStorageObject);
+    }
+
+    private void createRestoreSliceOnSidecar(BundleStorageObject bundleStorageObject)
+    {
+        if (writerContext.job().isCoordinatedWriteEnabled())
+        {
+            CoordinatedCloudStorageDataTransferApi coordinatedApi = (CoordinatedCloudStorageDataTransferApi) dataTransferApi;
+            coordinatedApi.forEach((clusterId, ignored) -> {
+                String readBucket = transportContext.transportConfiguration().readAccessConfiguration(clusterId).bucket();
+                CreateSliceRequestPayload slicePayload = toCreateSliceRequestPayload(bundleStorageObject, readBucket);
+                coordinatedApi.createRestoreSliceFromExecutor(clusterId, slicePayload);
+            });
+            // no need to add to the createdRestoreSlices. Driver polls the progress of the entire restore job.
+        }
+        else
         {
+            CreateSliceRequestPayload slicePayload = toCreateSliceRequestPayload(bundleStorageObject);
+            // Create slices on all replicas; remove the _failed_ replica if operation fails
+            replicas.removeIf(replica -> !tryCreateRestoreSlicePerReplica(replica, slicePayload));
+            // exit early if all replicas has failed, i.e. CL cannot be satisfied for sure.
+            if (replicas.isEmpty())
```

Like above, would it be possible to actually check the CL here and fail early?